src/plugins/reference/README.md

```diff
@@ -0,0 +1,128 @@
+- infos = Information about the template plugin is in keys below
+- infos/author = Klemens Böswirth <k.boeswirth+git@gmail.com>
+- infos/licence = BSD
+- infos/needs =
+- infos/provides =
+- infos/recommends =
+- infos/placements = presetstorage
+- infos/status = nodep libc preview experimental unfinished
+- infos/metadata = check/reference check/reference/restrict
+- infos/description = Useful plugin for validating singular or recursive references
+
+## Introduction
+
+This plugin serves one single purpose: the validation of reference values inside of keys.
+
+To specify a key as a reference add the metakey `check/reference`. Currently the metakey 
+supports the three values `single`, `recursive` and `alternative`. While the value 
+`alternative` only makes sense in a certain context inside the `recursive` mode, the other
+two values define to different modes of operation for the plugin. 
+
+If a key has the metakey `check/reference` with value `single`, the plugin reads the value
+of the key and produces an error, if the value is not a valid reference.
+
+If a key has metakey `check/reference` with value `recursive`, the plugin constructs a 
+reference graph starting with the marked key. (for the exact construction see below) 
+In this reference graph each node corresponds to a key in the KDB, while each edge 
+represents a reference between to keys. The plugin will produce an error, if this graph
+is not a directed acyclic graph or contains any invalid references.
+
+Additionally to this basic functionality the plugin allows one to restrict the set of valid
+references using the `check/reference/restrict` metakey. The exact mechanism for this is 
+described below.
+
+### Resolution of references
+If the key `A` is interpreted as a reference by the plugin, it has to be an array. Each of 
+the array elements must contain a so called reference value. This reference value will be 
+resolved into a keyname, which will then be validated. The resolution goes as follows:
+
+* If the reference value starts with `/`, it is treated as absolute and interpreted as a 
+  keyname directly.
+* Otherwise the reference value will be interpreted as relative to the **parent** of `A`.
+  That means a reference which does not contain a `/` will also was refer to a sibling of `A`,
+  the reference array key.
+* `.` and `..` will be treated the same way as in UNIX paths. If these are used in an absolute
+  reference, the plugin will emit a warning, because their use in this case is unnecessary and 
+  might be a mistake. The same is true, if `.` is used anywhere, but at the start of the 
+  reference.
+
+### Construction of the reference graph
+
+The process starts with a key `X`, which has the metakey `check/reference` set to the value
```

Comment:
Which one is the key X?

Reply:
`X` is the key which has `check/reference` set to `recursive`.

So in the case of `kdb setmeta user/tests/somekey/ref check/reference recursive` the key `user/tests/somekey/ref` will be `X`.